WEB-INF/classes/gov/noaa/pfel/erddap/jte/TableOptions.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  public String getOtherClasses() {
+    return (otherClasses == null || otherClasses.length() == 0 ? "" : " " + otherClasses);
+  }
+
+  public String getBgColor() {
+    return bgColor == null ? "false" : bgColor;
+  }
+
+  public int getNColumns() {
+    return table.nColumns();
+  }
+
+  public int getNRows() {
+    return table.nRows();
+  }
+
+  public String insideForLoophead(int col) {
+    String s = table.getColumnName(col);
+    if (needEncodingAsHtml) {
+      s = XML.encodeAsHTML(s);
+    }
+    if (!somethingWritten && s.trim().length() > 0) {
+      somethingWritten = true;
+    } else if (col == getNColumns() - 1 && s.trim().length() == 0) {
+      s = "&nbsp;";
+    }
+
+    Attributes catts = table.columnAttributes(col);
+    fileAccessBaseUrl[col] = catts.getString("fileAccessBaseUrl");
+    fileAccessSuffix[col] = catts.getString("fileAccessSuffix");
+    if (!String2.isSomething(fileAccessBaseUrl[col])) fileAccessBaseUrl[col] = "";
+    if (!String2.isSomething(fileAccessSuffix[col])) fileAccessSuffix[col] = "";
+
+    return s;
+  }
+
+  public boolean getWriteUnits() {
+    return writeUnits;
+  }
+
+  public String insideForLoopUnits(int col) {
+    String tUnits = table.columnAttributes(col).getString("units");
+    if (col == timeColumn) tUnits = "UTC";
+    if (tUnits == null) tUnits = "";
+    if (needEncodingAsHtml) tUnits = XML.encodeAsHTML(tUnits);
+    if (!somethingWritten && tUnits.trim().length() > 0) {
+      somethingWritten = true;
+    } else if (col == getNColumns() - 1 && tUnits.trim().length() == 0) {
+      tUnits = "&nbsp;";
+    }
+    return tUnits;
+  }
+
+  public int getTimeColumn() {
+    return timeColumn;
+  }
+
+  public String getDoubleDataConverted(int col, int row) {
+    return Calendar2.safeEpochSecondsToIsoStringTZ(table.getDoubleData(col, row), "");
+  }
+
+  public String getStringData(int col, int row) {
+    return table.getStringData(col, row);
+  }
+
+  public boolean checkLengthFile(int col) {
+    return (fileAccessBaseUrl[col].length() > 0 || fileAccessSuffix[col].length() > 0);
+  }
+
+  public String getTs(String s) {
+    return needEncodingAsHtml ? s : XML.decodeEntities(s);
+  }
+
+  public String getUrl(int col, String s) {
+    return fileAccessBaseUrl[col] + getTs(s) + fileAccessSuffix[col];
+  }
+
+  public String getEncodeashtml(String s) {
+    return needEncodingAsHtml ? XML.encodeAsHTML(s) : s;
+  }
+
+  public boolean checkcontainsurl(String s) {
+    return needEncodingAsHtml && String2.containsUrl(s);
+  }
+
+  public boolean checkmouseover(String s) {
+    return !needEncodingAsHtml
+        && !s.contains("href=")
+        && !s.contains("onmouseover")
+        && String2.containsUrl(XML.decodeEntities(s));
+  }
+
+  public List<String> getExtractUrls(String s) {
+    return String2.extractUrls(s);
+  }
+
+  public boolean getContainsUrl(String s) {
+    return String2.containsUrl(s);
+  }
+
+  public String encodeaddhttps(String s) {
```

Comment:
```suggestion
  public String encodeAddHttps(String s) {
```

Method name case fixing